agents/agents-features/agents-features-sql/src/jvmTest/kotlin/ai/koog/agents/features/sql/providers/SQLPersistenceProvidersTest.kt

```diff
@@ -0,0 +1,167 @@
+package ai.koog.agents.features.sql.providers
+
+import ai.koog.agents.snapshot.feature.AgentCheckpointData
+import ai.koog.prompt.message.Message
+import ai.koog.prompt.message.RequestMetaInfo
+import ai.koog.prompt.message.ResponseMetaInfo
+import kotlinx.coroutines.runBlocking
+import kotlinx.datetime.Clock
+import kotlinx.serialization.json.JsonPrimitive
+import kotlin.test.*
+
+/**
+ * Tests for SQL persistence providers.
+ * 
+ * Focuses on H2 in-memory database for actual functionality tests
+ * and compilation verification for other providers.
+ */
+class SQLPersistenceProvidersTest {
```

Comment:
I see testcontainers in test dependencies, but I don't see them being used. Let's create a number of test suites, one per each dialect provider. This should be relatively simple to do with testcontainers. 

We can extract common test logic with the tests themsevles to the abstract class, and then implementing test suites would only need to provide the correct storage provider implementation (Postgres, MySQL, etc.) and handle it's creation and shutdown.